### ISSUE TITLE
Members Page

### ISSUE THREAD

**Issue by brianking** [EclipseFdn/ecdtools.eclipse.org#52]
I think it would benefit the site by having a dedicated page / section on the site for members.

Currently I am thinking the changes would be:

* In the Members section on the home page, list fewer members and add a link/button to go to the Members page that will list all members.
* Change the Members link on the top-level menu to go to the Members page
* Move Testimonials from the home page to the Members page

**Comment by autumnfound** [comment]
@brianking Please see https://deploy-preview-54--eclipsefdn-ecdtools.netlify.app/membership/ for an initial rough draft of content. Right now the page is very bare, would you want to add more content to the page? Two samples of what we've done in the past can be seen on the [Jakarta EE page](https://jakarta.ee/membership/) and [IoT page](https://iot.eclipse.org/membership/).

Additionally, we have no out of the box way of displaying some members and having them randomize on page load (which is the best way to do it to ensure there's no favouritism). We have the functionality on IoT but would need to port it to the main theme and look at how to integrate it with the rest of our components. Is this something that's needed on the homepage? The sites we have in the portfolio have mixed implementations for this which is why I ask.

**Comment by brianking** [comment]
I like it so far, thanks @autumnfound. I don't think the page is that much more bare, when compared to the other WG member pages you link to. So for me it is good to go. What I would add straight away however is:

* The 'Contact Us About Membership' button somewhere at the top of the page, pointing to https://accounts.eclipse.org/contact/membership/ecdtools
* Add a Benefits section. I can provide the content for that. We don't have a prospectus yet, but there are some links we can add at the end of the benefits section including to the charter: https://www.eclipse.org/org/workinggroups/eclipse_cloud_development_charter.php

If you think the page would benefit from it, we could add a right sidebar like on the other WG pages. We'd have to figure out what we want to put there.

Randomization is not important right now, we'll take it when it is ready in the main theme.

**Comment by autumnfound** [comment]
@brianking I've made a couple of updates, to the homepage and the membership page. WDYT?

**Comment by brianking** [comment]
Looks good. A few things:

1. On the homepage, change the section title from 'Founding Members' to just 'Members'.
2. Do you think the page would benefit from a taller header like the homepage?
3. We can add some bullet points at the top about membership
4. We can add a section about benefits also with bullet points

Feel free to add placeholder text for 3&4 until I am able to pull together some content. Then we'll be ready for deployment!

**Comment by autumnfound** [comment]
We typically use the larger top section or jumbotron when a page acts as a landing page. We might be able to make it work by linking to the different sections on the page, as well as contact us, but it's stretching it at the moment. Otherwise, I've made updates

**Comment by brianking** [comment]
First pass at content, this will need to be reviewed and iterated on. I will let you know when it is ready:
https://docs.google.com/document/d/1JiTWvjEFWOZSAYPVDuogsxY_UZhJA1jo-SCpIFjDobs/edit?usp=sharing

**Comment by brianking** [comment]
@autumnfound We have the content to proceed with this one. 
https://docs.google.com/document/d/1JiTWvjEFWOZSAYPVDuogsxY_UZhJA1jo-SCpIFjDobs/edit?usp=sharing

Requested changes:

* Add introductory content from document
* Updates the benefits section based on section in doc
* Remove the 'Contact us about membership' section. Move the button in that section up to the top somewhere, and change the text to 'Become a member'
* Change the section title from 'Founding Members' to 'Members'
* Update member list to current (see below)
* Change testimonials based on section in the document. We have made edits for them to be shorter

Current members: 

<img width="935" alt="Screenshot 2020-09-25 at 17 40 24" src="https://user-images.githubusercontent.com/114116/94288181-6b8f1c00-ff57-11ea-9705-a0b18f1c6039.png">

**Comment by autumnfound** [comment]
@brianking Since we've broken away from founding members as the only grouping of members, how should we display members on the home page?

**Comment by autumnfound** [comment]
Additionally, I've updated https://deploy-preview-54--eclipsefdn-ecdtools.netlify.app/membership/  with new content. Feel free to check it and let me know what you think

**Comment by brianking** [comment]
@autumnfound How about presenting the members in the categories presented in the image in the comment earlier here?

**Comment by autumnfound** [comment]
On the homepage? That would take up a lot of screen real estate and reduce the use and traffic to the page we are creating here. Do we still want to have a random selection of the available members/3 of them displayed with the View All Members button? Do we want to just remove the section and have the link in the header?

**Comment by croundy** [comment]
I would like the look and feel to be a bit warmer and visually inviting.  Even if we mirror the IoT page for now https://iot.eclipse.org/membership/

1) Larger top banner with a "contact us about membership" button that links to the membership form
2) Rounded vs. square buttons, which would be a site wide issue
3) More separation between buttons
4) Cleaner member testimonial section
5) Right side navigation to take visitors to the most important content 
6) Consider a separate page highlighting members. Similar to this: https://iot.eclipse.org/membership/members/

**Comment by brianking** [comment]
@autumnfound Sorry I misunderstood. For the hompage, let's have it like https://iot.eclipse.org/ and just highlight use any 3 Strategic members.

'Become a member' should go to https://accounts.eclipse.org/contact/membership/ecdtools (it should also go on top of the Members page) and 'View all members' should go to the new page.

I agree mostly with Clark's suggestions, but we don't need a separate page highlighting members for the first pass. We can add it later

**Comment by croundy** [comment]
agree that we can add the members page later...just want to make sure it's on the radar

**Comment by autumnfound** [comment]
I've added changes for most of the feedback, except the following:

> 4. Cleaner member testimonial section

How would you clean this section up? I think something we could do is darken the background image of the section by 33%-50% to make the text much more clear, and make the organization names larger and more prominent.

> 5. Right side navigation to take visitors to the most important content

Because of the current page structure, adding a sidebar would be mighty complicated with the current implementation. We should split this out into its own issue to discuss outside the scope of this issue.

**Comment by brianking** [comment]
New updates look good.

Agreed in splitting the structure out to a separate issue, the first version of this that goes live can focus just on the content.

**Comment by croundy** [comment]
Thank you. Definitely an improvement and moving in the right direction.  To your questions:

4. Member testimonials. 

-  The background is rather busy and competes with the logos and text. Perhaps change to just a solid background

- Change "Testimonials" to something less formal, such as "What Our Members are Saying"

- I don't know that the company names below the logos are necessary. Perhaps remove them

- We did have names and titles attached to each quote. Is there a reason those are not included

In the future I would like to see quotes show up dynamically (1 at a time) complete with a nice profile picture/image of the member, along with the company logo. 

5.  Right side navigation. 

- Understand and agree  


'

**Comment by croundy** [comment]
For the Jumbotron, if possible to give it similar look and feel to the IoT membership page, it would give that space more polish https://iot.eclipse.org/membership/

My thoughts:

- Change "Membership" to "Join Us"

- Move this statement from the body text to the Jumbotron: "Join industry leaders like Ericsson, IBM, Red Hat, and SAP as we define and build the future of Web and cloud development tools"

- Add a "Membership" headline above the first paragraph of text

**Comment by autumnfound** [comment]
> The background is rather busy and competes with the logos and text. Perhaps change to just a solid background

This is why I suggested making the background image darker/more muted. We use images to make the section pop more than just a plain coloured section while making the content more prominent.

> We did have names and titles attached to each quote. Is there a reason those are not included

Previous mark up didn't have it, and I haven't updated the actual component to include the names yet. Apologies!

> In the future I would like to see quotes show up dynamically (1 at a time) complete with a nice profile picture/image of the member, along with the company logo. If there is any way to use the same template as the IoT page, it would clean things up significantly:

There is a way we can make it work through a carousel that would nicely bridge the dynamic implementation and the look and feel. I'll speak to Chris about creating a new component or look and feel for this!

> For the Jumbotron, if possible to give it similar look and feel to the IoT membership page, it would give that space more polish https://iot.eclipse.org/membership/
> 
> My thoughts:
> 
> * Change "Membership" to "Join Us"
> * Move this statement from the body text to the Jumbotron: "Join industry leaders like Ericsson, IBM, Red Hat, and SAP as we define and build the future of Web and cloud development tools"
> * Add a "Membership" headline above the first paragraph of text

For the jumbotron styling, that is a global styling on the IoT site. We currently have no way of easily applying that to a single page, and doing so isn't advised. Having inconsistent styling across a site can be jarring or distracting.

I can update the tagline to be that text, but I would have to remove the links as they aren't supported in that section.

**Comment by croundy** [comment]
Thank you for all the help! About the global styling, does it make sense to use it across the entire ECD site? It does seem cleaner and more polished that what we are currently using.

**Comment by autumnfound** [comment]
We should split that out as a separate request. I agree its a cleaner look overall, but if you don't have something in the design to fill the whitespace, then it makes it feel more empty.

**Comment by autumnfound** [comment]
I've updated the sample site  https://deploy-preview-54--eclipsefdn-ecdtools.netlify.app/ with the new changes. The changes in this patch impact more than just the membership page now, but all of the changes are for the better IMO. The noticeable changes; headings are no longer forced to be uppercase, and buttons are now pills. 

On the membership page, I've fixed testimonials title, people's name, title, and workplace, and made the image darker which has done the trick neatly I think! I've moved the first sentence to the jumbotron (minus links), and updated the titles. WDYT?

**Comment by croundy** [comment]
Definitely an improvement. Thank you! 

For the testimonials, I'm thinking that we should actually make the names and titles smaller than the quotes. And perhaps even italicized or some other treatment to set them apart from the quotes themselves.

**Comment by croundy** [comment]
Now reviewing the top nav, am thinking we can eliminate "Join" and "Testimonials" which will give us room to add "Projects" and "Adopters" as well as social media icons to make them more visible

**Comment by brianking** [comment]
@autumnfound I notice the Members **section** on the homepage is gone now. Intentional?

**Comment by chrisguindon** [comment]
> @autumnfound I notice the Members **section** on the homepage is gone now. Intentional?

I see it..

**Comment by autumnfound** [comment]
Updated to add members section back (removed it as it was breaking while I was playing around with the members), and updated the navigation. There are still convos to be had around how to handle social icons, but that has its own issue already!

**Comment by autumnfound** [comment]
@croundy @brianking so final look through, are you happy with the current look and feel of the members page?

**Comment by brianking** [comment]
Looks good to me!

**Comment by croundy** [comment]
Definitely a big improvement. Many thanks for all your good work! One minor thing: 

1. We have "Web" capitalized on the tagline. Let's make that lower case

Next steps: would like to explore the other improvements we discussed

- Dynamic member quotes

- Social media links to the top nav

- Right side nav 

We will also need to add top nav for some new pages (adopters, projects, etc,) when content is ready. Thanks again!

**Comment by brianking** [comment]
I'm ok with deploying what we have, and opening other issues for the rest of the enhancements.

**Comment by chrisguindon** [comment]
@brianking @croundy 

Ii will send you both a meeting invite. I think we need to discuss a better way for us to collaborate on these projects. 

I think it would be best for you to work with a designer before we start implementing these new pages.

**Comment by chrisguindon** [comment]
As discussed, we can go live with this. @croundy and @brianking will create new issues for the remaining tasks.
